build.gradle

```diff
@@ -882,6 +882,7 @@ project(':server') {
     inputs.property "commitId", commitId
     inputs.property "version", version
     outputs.file receiptFile
+    outputs.cacheIf { true }
```

Comment:
That means the jars could have old `kafka-version.properties` when its module has no change. This change is for CI, right? we want to cache the jar if no source files change. Maybe we can enable the `cacheIf` when the env `GITHUB_ACTIONS` exists. Otherwise, we keep origin behavior. WDYT?

Reply:
`outputs.cacheIf` is one way to tell Gradle that a task is cacheable https://docs.gradle.org/current/userguide/build_cache.html#using_the_runtime_api.

Even with this declaration, the task will only be up-to-date / cacheable if the inputs are unchanged. 

So, taking the previous task definition:

```
task createVersionFile() {
    inputs.property "commitId", commitId 
    inputs.property "version", version
    outputs.file receiptFile
```

For incremental builds, this task will be considered UP-TO-DATE as long as the two inputs don't change. In order to actually allow this output to be _cached_, we need to tell gradle that it's cacheable. That's done by adding the `@CacheableTask` for custom task classes, or by adding `outputs.cacheIf { true }`